lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -777,6 +783,10 @@ namespace {
 /// unsigned as seen from the Verilog language perspective.
 enum SubExprSignResult { IsSigned, IsUnsigned };
 
+/// This enum keeps track of whether the emitted subexpression was a constant,
+/// a wire, or anything more complex (like a parenthesized expression).
+enum class SubExprKind { Const, Wire, Other };
```

Comment:
Please don't add this as logic in the printer itself.  Instead, just add it to the prepass in this file, similar to how we break down variadic ops into binary ones.

Reply:
Good point! Reverted all the `SubExprKind` business and am now doing things as a pre-pass.